iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java

```diff
@@ -135,57 +165,79 @@ private void doRemoveDataNode(String[] args)
    */
   private List<TDataNodeLocation> buildDataNodeLocations(String args) {
     List<TDataNodeLocation> dataNodeLocations = new ArrayList<>();
-    if (args == null || args.trim().isEmpty()) {
-      return dataNodeLocations;
-    }
 
     // Now support only single datanode deletion
     if (args.split(",").length > 1) {
-      LOGGER.info("Incorrect input format, usage: <id>/<ip>:<rpc-port>");
-      return dataNodeLocations;
+      throw new IllegalArgumentException("Currently only removing single nodes is supported.");
     }
 
     // Below supports multiple datanode deletion, split by ',', and is reserved for extension
-    try {
-      List<TEndPoint> endPoints = NodeUrlUtils.parseTEndPointUrls(args);
-      try (ConfigNodeClient client =
-          ConfigNodeClientManager.getInstance().borrowClient(ConfigNodeInfo.CONFIG_REGION_ID)) {
-        dataNodeLocations =
-            client.getDataNodeConfiguration(-1).getDataNodeConfigurationMap().values().stream()
-                .map(TDataNodeConfiguration::getLocation)
-                .filter(location -> endPoints.contains(location.getClientRpcEndPoint()))
-                .collect(Collectors.toList());
-      } catch (TException | ClientManagerException e) {
-        LOGGER.error("Get data node locations failed", e);
+    List<NodeCoordinate> endPoints = parseCoordinates(args);
+    try (ConfigNodeClient client =
+        configNodeClientManager.borrowClient(ConfigNodeInfo.CONFIG_REGION_ID)) {
+      dataNodeLocations =
+          client.getDataNodeConfiguration(-1).getDataNodeConfigurationMap().values().stream()
+              .map(TDataNodeConfiguration::getLocation)
+              .filter(
+                  location ->
+                      endPoints.stream()
+                          .anyMatch(nodeCoordinate -> nodeCoordinate.matches(location)))
+              .collect(Collectors.toList());
+    } catch (TException | ClientManagerException e) {
+      LOGGER.error("Get data node locations failed", e);
+    }
+
+    return dataNodeLocations;
+  }
+
+  protected List<NodeCoordinate> parseCoordinates(String coordinatesString) {
+    // Multiple coordinates are separated by ","
```

Comment:
Looks like it currently supports deleting a datanode node? So the judgment here seems superfluous? 
Of course, it's a good design if it's for future scalability

Reply:
I think it's OK for now